dotnet/samples/HomeAutomation/Program.cs

```diff
@@ -0,0 +1,93 @@
+﻿/*
+ Copyright (c) Microsoft. All rights reserved.
+
+ Example application that uses Semantic Kernel.
+
+ Loads app configuration from:
+ - appsettings.json.
+ - appsettings.{Environment}.json.
+ - Secret Manager when the app runs in the "Development" environment (set through the DOTNET_ENVIRONMENT variable).
+ - Environment variables.
+ - Command-line arguments.
+*/
+
+using HomeAutomation.Plugins;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.ChatCompletion;
+using Microsoft.SemanticKernel.Connectors.OpenAI;
+
+namespace HomeAutomation;
+
+internal static class Program
+{
+    internal static async Task Main(string[] args)
+    {
+        HostApplicationBuilder builder = Host.CreateApplicationBuilder(args);
+        builder.Services.AddHostedService<Worker>();
+        builder.Services.AddSingleton<AppConfiguration>();
+        builder.Services.AddSingleton<Kernel>(sp =>
+        {
+            AppConfiguration ac = sp.GetRequiredService<AppConfiguration>();
+            var builder = Kernel.CreateBuilder()
+                                // Add AI services here
+                                // Note that you can provide your custom http client when adding specific AI services
+                                .AddAzureOpenAIChatCompletion(ac.AzureOpenAiDeployment, ac.AzureOpenAiEndpoint, ac.AzureOpenAiApiKey);
+            // Add plugins to include in kernel here
+            builder.Plugins.AddFromType<MyTimePlugin>();
```

Comment:
If you register the kernelBuilder with `builder.Services.AddKernel();` then the plugins can be registered as singletons separately from the kernel and will be reused even if the kernel itself is transient.

Reply:
Did something equivalent by moving all registrations to the app DI container instead